### PR TITLE
Members Command - MF 5 Implementation

### LINKED ISSUE #1431: Implement the members command.

### PR DESCRIPTION
**Pull request by dmccoystephenson**
The info command will send the player a list of the members in their faction. Eventually we will want to allow viewing the members of other factions.

closes #1431

### CODE REVIEW

**Review by alyphen — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/dansplugins/factionsystem/command/faction/MfFactionCommand.kt [hunk 1]
```diff
@@ -10,6 +10,8 @@ import com.dansplugins.factionsystem.command.faction.join.MfFactionJoinCommand
 import com.dansplugins.factionsystem.command.faction.law.MfFactionLawCommand
 import com.dansplugins.factionsystem.command.faction.makepeace.MfFactionMakePeaceCommand
 import com.dansplugins.factionsystem.command.faction.help.MfFactionHelpCommand
+import com.dansplugins.factionsystem.command.faction.members.MfFactionMembersCommand
+import com.dansplugins.factionsystem.faction.MfFactionMember
```
[the remark this answers is not in the document]
Comment (dmccoystephenson) [reply src/main/kotlin/com/dansplugins/factionsystem/command/faction/MfFactionCommand.kt:14]: Nope! Good catch!

[the remark this answers is not in the document]
Comment (dmccoystephenson) [reply src/main/kotlin/com/dansplugins/factionsystem/command/faction/MfFactionCommand.kt]: This has been removed.

File: src/main/kotlin/com/dansplugins/factionsystem/command/faction/members/MfFactionMembersCommand.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            val faction = factionService.getFaction(mfPlayer.id)
+            if (faction == null) {
+                sender.sendMessage("$RED${plugin.language["CommandFactionMembersMustBeInAFaction"]}")
+                return@Runnable
+            }
+            // send player list of members
+            sender.sendMessage("$AQUA${plugin.language["CommandFactionMembersTitle", faction.name]}")
+            for (member in faction.members) {
+                sender.sendMessage("$AQUA" + member.player.toBukkit().getName())
+            }
```
[the remark this answers is not in the document]
Comment (dmccoystephenson) [reply src/main/kotlin/com/dansplugins/factionsystem/command/faction/members/MfFactionMembersCommand.kt:44]: What does your suggested implementation do? Does it print it with no new lines?

[the remark this answers is not in the document]
Comment (alyphen) [reply src/main/kotlin/com/dansplugins/factionsystem/command/faction/members/MfFactionMembersCommand.kt:44]: Breaks members into groups of 20 and then prints up to 20 members per line

File: src/main/kotlin/com/dansplugins/factionsystem/command/faction/members/MfFactionMembersCommand.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                    sender.sendMessage("$RED${plugin.language["CommandFactionMembersFailedToSavePlayer"]}")
+                    plugin.logger.log(SEVERE, "Failed to save player: ${it.reason.message}", it.reason.cause)
+                    return@Runnable
+                }
+            val factionService = plugin.services.factionService
+            val faction = factionService.getFaction(mfPlayer.id)
+            if (faction == null) {
+                sender.sendMessage("$RED${plugin.language["CommandFactionMembersMustBeInAFaction"]}")
+                return@Runnable
+            }
```
[the remark this answers is not in the document]
Comment (dmccoystephenson) [reply src/main/kotlin/com/dansplugins/factionsystem/command/faction/members/MfFactionMembersCommand.kt:39]: I've added a role permission here. Please let me know if I've done it correctly.

[the remark this answers is not in the document]
Comment (alyphen) [reply src/main/kotlin/com/dansplugins/factionsystem/command/faction/members/MfFactionMembersCommand.kt:39]: Role permissions also need to be added to:
* MfFactionPermission.kt: permission constant, valueOf method for lookup, and values method for retrieving all values
* MfFaction.kt: default permission values
* MfFactionRoles.kt: default role settings

The constant from MfFactionRoles can then be imported.
I tend to write permissions there with action words - i.e. "Change name", "Add law", "Set flag", "View info", "Chat" - so for this one I think "VIEW_MEMBERS" would likely be a better name than "MEMBERS"

[the remark this answers is not in the document]
Comment (dmccoystephenson) [reply src/main/kotlin/com/dansplugins/factionsystem/command/faction/members/MfFactionMembersCommand.kt:40]: Thank you for this! The LIST_MEMBERS role permission should be all set now.

File: src/main/kotlin/com/dansplugins/factionsystem/command/faction/members/MfFactionMembersCommand.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmccoystephenson) [reply src/main/kotlin/com/dansplugins/factionsystem/command/faction/members/MfFactionMembersCommand.kt:56]: This should now print all of the members on the same line.

[the remark this answers is not in the document]
Comment (alyphen) [reply src/main/kotlin/com/dansplugins/factionsystem/command/faction/members/MfFactionMembersCommand.kt:56]: Perhaps ```faction.members.joinToString { it.player.toBukkit().name }``` would be more succinct?

File: src/main/kotlin/com/dansplugins/factionsystem/command/faction/members/MfFactionMembersCommand.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (alyphen) [reply src/main/kotlin/com/dansplugins/factionsystem/command/faction/members/MfFactionMembersCommand.kt:40]: Nice job :)

File: src/main/kotlin/com/dansplugins/factionsystem/command/faction/members/MfFactionMembersCommand.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmccoystephenson) [reply src/main/kotlin/com/dansplugins/factionsystem/command/faction/members/MfFactionMembersCommand.kt]: This has been made more succinct.

**Review by alyphen — APPROVED** [review]

**Pull request merged**